### PR TITLE
Fixed Legend position option in the Line graph > slope option

### LINKED ISSUE #9196: Confusion over Legend especially on Line Plot > Slope
Hello @rdstern, @N-thony and @derekagorhom, 

Please tell me if I am misunderstanding this. 

I notice that there are two legend check boxes on the Line Plot > Slope Tab. Is this an error?  They both seem related to the same item but I am unclear how they interact. 
When both are checked the lower seems to define the location. 

If I uncheck the lower then I still have a legend but it appears on the right hand side. 

But then if I uncheck the upper and put the lower on the right it disappears.

Then when i make a heatmap with the single legend box unchecked I get a legend. Perhaps having a legend is the default and the checkbox is for specifying the location or omitting it? If so please confirm, is this for all graphs? Then I will write the help as such. 

Thanks, 

Rachel

### PR DESCRIPTION
**Pull request by derekagorhom**
Fixes #9196 
@rachelkg , @N-thony , @rdstern I have disabled the lower legend position checkbox when the user is in the Slope option of the line graph dialogue. I have added a drop down input position to the legend option at the top. 
Can you review this PR
Thanks

### CODE REVIEW

**Comment by rdstern** [comment]
@derekagorhom I hope the solution is simpler, namely delete the top legend control.  

We added the lower control to many graph dialogs, and I assume we did not notice it has an inferior version there.  Can you try that anyway.

**Comment by derekagorhom** [comment]
@rdstern I have made so that the lower legend option work for the slope option now and deleted the top option...can you have a look
Thanks

**Review by rdstern — CHANGES_REQUESTED** [review]
@derekagorhom can you also fix the single Legend control.  

By the way, the greenrice data is in the agritutorial package if you want to use the same data as in the issue. 
Start by making W and N into factors.  You could use the slope (as Rachel does) or start with the line plot - the graph is silly there, but that's not the point. 
Start with the dialog as so:

image

Notice I haven't done anything to the legend, so it is unchecked.

Press Ok.  Then get the dialog back and the legend control is checked.  That shouldn't happen.  Note I checked with the latest version and the control is ok there.  So, this is an example where you have fixed one fault and now introduced a new problem in that dialog, that wasn't there before.  And this new problem affects all the 5 options at the top of the dialog.  Please fix, and maybe use this example of a "regression in the code" to check that you don't do this sort of thing too often.

**Comment by derekagorhom** [comment]
> @derekagorhom can you also fix the single Legend control.
> 
> By the way, the greenrice data is in the agritutorial package if you want to use the same data as in the issue. Start by making W and N into factors. You could use the slope (as Rachel does) or start with the line plot - the graph is silly there, but that's not the point. Start with the dialog as so:
> 
> image
> 
> Notice I haven't done anything to the legend, so it is unchecked.
> 
> Press Ok. Then get the dialog back and the legend control is checked. That shouldn't happen. Note I checked with the latest version and the control is ok there. So, this is an example where you have fixed one fault and now introduced a new problem in that dialog, that wasn't there before. And this new problem affects all the 5 options at the top of the dialog. Please fix, and maybe use this example of a "regression in the code" to check that you don't do this sort of thing too often.

@rdstern I have fixed the problem you identified. 
can you have a look again 
Thanks

**Review by rdstern — APPROVED** [review]
Thanks @derekagorhom that looks fine now.  @N-thony over to you

**Comment by rdstern** [comment]
@N-thony over to you

File: instat/dlgLinePlot.vb [hunk 1]
```diff
@@ -858,9 +855,22 @@ Public Class dlgLinePlot
...
+                clsThemeFunction.AddParameter("legend.position", Chr(34) & ucrInputLegendPosition.GetValueToSet & Chr(34), iPosition:=1)
+                clsBaseOperator.RemoveParameterByName("c")
+            Else
+                clsBaseOperator.RemoveParameterByName("theme")
+                clsBaseOperator.RemoveParameterByName("slopetheme")
+            End If
+        End If
+        If rdoLine.Checked AndAlso rdoLinerange.Checked AndAlso rdoSmoothing.Checked AndAlso rdoDumbbell.Checked Then
+            AddRemoveTheme()
+        End If
```
[the remark this answers is not in the document]
Comment (derekagorhom) [reply instat/dlgLinePlot.vb]: I changed it from `AndAlso` to `OrElse`

File: instat/dlgLinePlot.vb [hunk 2]
```diff
@@ -1278,4 +1272,8 @@ Public Class dlgLinePlot
     Private Sub ucrChkAddLineLineRange_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkAddLineLineRange.ControlValueChanged
         AddRemoveLineOnLineRange()
     End Sub
+
+    Private Sub AllControl_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrVariablesAsFactorForLinePlot.ControlContentsChanged, ucrSave.ControlContentsChanged, ucrReceiverYMin.ControlContentsChanged, ucrReceiverYMax.ControlContentsChanged, ucrReceiverXEnd.ControlContentsChanged, ucrReceiverX.ControlContentsChanged, ucrReceiverSlopeY.ControlContentsChanged, ucrReceiverSlopeX.ControlContentsChanged, ucrReceiverSlopeColour.ControlContentsChanged, ucrFactorOptionalReceiver.ControlContentsChanged
+
+    End Sub
```
[the remark this answers is not in the document]
Comment (derekagorhom) [reply instat/dlgLinePlot.vb]: I have removed this sub, i think its from me making some changes in the designer

File: instat/dlgLinePlot.vb (hunk 1 above)
[the remark this answers is not in the document]
Comment (derekagorhom) [reply instat/dlgLinePlot.vb]: No... these are the other options in the dialog

**Comment by derekagorhom** [comment]
@rdstern can review this again

**Review by rdstern — APPROVED** [review]
@N-thony back to you

**Pull request merged**